independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/injection/assignability/generics/AssignabilityWithGenericsTest.java

```diff
@@ -189,4 +196,83 @@ public String ping(T unusedParam) {
 
     }
 
+
+
+
+    // This succeeds
+
+    @Test
+    public void testOptaPlanner_object() {
+        InstanceHandle<ScoreManager1<MyScore1>> instance = Arc.container().instance(
+                new TypeLiteral<ScoreManager1<MyScore1>>() {});
+        assertTrue(instance.isAvailable());
+    }
+
+    static class ScoreManager1<S> {}
+
+    static class MyScore1 {}
+
+    static class ScoreManagerBeanProvider1 {
+
+        @DefaultBean
+        @Singleton
+        @Produces
+        // SUCCESS: This matches with ScoreManager1<MySolution>
+        <S> ScoreManager1<S> scoreManager() {
+            return new ScoreManager1<>();
+        }
+    }
+
+    // This succeeds
+
+    @Test
+    public void testOptaPlanner_score_untyped() {
+        InstanceHandle<ScoreManager2<MyScore2>> instance = Arc.container().instance(
+                new TypeLiteral<ScoreManager2<MyScore2>>() {});
+        assertTrue(instance.isAvailable());
+    }
+
+    static class ScoreManager2<S extends Score2> {}
+
+    interface Score2 {}
+
+    static class MyScore2 implements Score2 {}
+
+    static class ScoreManagerBeanProvider2 {
+
+        @DefaultBean
+        @Singleton
+        @Produces
+        // SUCCESS: This matches with ScoreManager2<MyScore2>
+        <S extends Score2> ScoreManager2<S> scoreManager() {
+            return new ScoreManager2<>();
+        }
+    }
+
+    // This fails
+
+//    @Test
+//    public void testOptaPlanner_score_typed_too() {
+//        InstanceHandle<ScoreManager3<MyScore3>> instance = Arc.container().instance(
+//                new TypeLiteral<ScoreManager3<MyScore3>>() {});
+//        assertTrue(instance.isAvailable());
+//    }
+//
+//    static class ScoreManager3<S extends Score3<S>> {}
+//
+//    interface Score3<S extends Score3<S>> {}
```

Comment:
This is uses generics in the same way that Enum does. This is the only difference with the lines above that do succeed. This causes ARC to slip up as far as I call tell.